### PR TITLE
Limb darkening function

### PR DESCRIPTION
**Pull request by gyengen**
Calculate limb darkening function for specified wavelength and use to remove limb darkening effects from input image.

### CODE REVIEW

**Comment by larrymanley** [comment]
I've tried this code with an HMI continuum image (6173 Å) (AIA 4500 is defunct) and it doesn't seem to work.  I tried changing it to use a temporary data variable and returning a new map and that doesn't work either.  There is a small difference in the mean - 33686.473 (before) vs. 33686.46220 (after).

File: sunpy/physics/transforms/limb_darkening_correct.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+
+def limb_darkening_correct(observation, limb_cut=True):
+    """Calculate limb darkening function for specified wavelength
+    and use to remove limb darkening effects from input image.
+    The limb darkening function uses a 5th ordern polynomial
+    fitting to the limb darkening constants obtained from
+    Astrophysical Quantities by Allan. Only in the wavelength
+    range between 4000 and 15000 A.
+
```
Comment (gyengen) [inline sunpy/physics/transforms/limb_darkening_correct.py]: Good idea, i will do soon. Cheers.

**Comment by gyengen** [comment]
Interesting. I test with HMI continuum image and it works for me. Anyway, it should be a significant difference between the input and output mean. Could you please tell me the date of the observation?

File: sunpy/physics/transforms/limb_darkening_correct.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    ---------
+        Cox, Arthur N., Allen's astrophysical quantities, 4th ed. Publisher:
+            New York: AIP Press; Springer, 2000. ISBN: 038798746
+
+    Examples
+    --------
+        >>> myfit = sunpy.map.Map('test.fits')
+        >>> myfit_c = limb_darkening_correct(myfit, limb_cut=True)"""
+    
+    wavelength = observation.meta['WAVELNTH']
```
Comment (Cadair) [inline sunpy/physics/transforms/limb_darkening_correct.py]: this should be `map.measurement`

File: sunpy/physics/transforms/limb_darkening_correct.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    x_center = observation.meta['crpix1']
+    y_center = observation.meta['crpix2']
```
Comment (Cadair) [inline sunpy/physics/transforms/limb_darkening_correct.py]: I assume here what you want is the pixel value for the centre of the Sun, which is always the 'CRPIX' so what you want is:

```python
x_centre, y_centre = map.data_to_pixel((0,0)*u.arcsec)
```

File: sunpy/physics/transforms/limb_darkening_correct.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    radius = observation.meta['RSUN_OBS'] / observation.meta['CDELT1']
```
Comment (Cadair) [inline sunpy/physics/transforms/limb_darkening_correct.py]: This should be `map.rsun_obs / map.scale[0]`

File: sunpy/physics/transforms/limb_darkening_correct.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    x_dim = observation.meta['NAXIS1']
+    y_dim = observation.meta['NAXIS2']
```
Comment (Cadair) [inline sunpy/physics/transforms/limb_darkening_correct.py]: this should be `x_dim, y_dim = map.dimensions`

**Comment by Cadair** [comment]
also I would suggest you move this file to `sunpy/physics/limb_darkening.py`. Also it is going to need a couple of tests.

**Comment by larrymanley** [comment]
I used hmi.ic_45s.2014.03.29_16_01_30_TAI.continuum.fits

**Comment by larrymanley** [comment]
I also noticed the limb_cut takes four minutes to run.  Recommend using the code below that Drew came up with - looks a little clunky but it's vectorized and runs instantly.

```
# remove HMI magnetogram off-limb noise
# thanks Drew Leonard!
lowx, highx = (hmimap.xrange[0].value / hmimap.scale.x.value,
               hmimap.xrange[1].value / hmimap.scale.x.value)
lowy, highy = (hmimap.yrange[0].value / hmimap.scale.y.value,
               hmimap.yrange[1].value / hmimap.scale.y.value)
x_grid, y_grid = np.mgrid[lowx:highx-1, lowy:highy-1]
r_grid = np.sqrt((x_grid ** 2.0) + (y_grid ** 2.0))
outer_rad = (hmimap.rsun_obs.value * 1.0) / hmimap.scale.x.value
hmimap.data[r_grid > outer_rad] = 0.0

```

File: sunpy/physics/limb_darkening.py [hunk 6]
```diff
@@ -35,46 +37,47 @@ def limb_darkening_correct(observation, limb_cut=True):
...
         >>> mymap = sunpy.map.Map('test.fits')
-        >>> limb_darkening_correct(mymap, limb_cut=True)"""
+        >>> newmap =limb_darkening_correct(mymap, limb_cut=True)
+                >>> mymap.peek()
+        >>> newmap.peek()"""
 
-    wavelength = observation.wavelength.value
 
-    if wavelength < 4000 or wavelength > 15000:
+    if observation.wavelength.value < 4000 or observation.wavelength.value > 15000:
```
Comment (Cadair) [inline sunpy/physics/limb_darkening.py]: This comparison should use quantities on both sides, not `.value`

**Comment by nabobalis** [comment]
Well I renamed the function and removed the tabs. I assume this was written before SunPy forced everything to use units. That will be changed. 

Do we want to put in a map and get out a map? Shouldn't this just work on an array?

**Comment by nabobalis** [comment]
Also this who;e github review system is janky as hell.

**Comment by nabobalis** [comment]
Closed in favour of #2344

**Pull request closed without merging**